numpy/lib/tests/test_nanfunctions.py

```diff
@@ -388,6 +410,101 @@ def test_empty(self):
             assert_equal(res, tgt)
 
 
+class TestNanFunctions_CumSumProd(TestCase, SharedNanFunctionsTestsMixin):
+
+    nanfuncs = [np.nancumsum, np.nancumprod]
+    stdfuncs = [np.cumsum, np.cumprod]
+
+    def test_allnans(self):
+        # Check for FutureWarning
+        for f, tgt_value in zip(self.nanfuncs, [0, 1]):
+            with warnings.catch_warnings(record=True) as w:
```

Comment:
It's cleaner to use [`assert_warns`](https://github.com/numpy/numpy/blob/a6efa09837b3a71eb6f6450b7dcf08c7381e75d0/numpy/testing/utils.py#L1683) and `assert_no_warnings` instead, e.g.,

```
with assert_warns(FutureWarning):
    # ....
```